src/ILLink.CodeFix/UnconditionalSuppressMessageCodeFixProvider.cs

```diff
@@ -0,0 +1,103 @@
+
+		public sealed override FixAllProvider GetFixAllProvider ()
+		{
+			// See https://github.com/dotnet/roslyn/blob/master/docs/analyzers/FixAllProvider.md for more information on Fix All Providers
+			return WellKnownFixAllProviders.BatchFixer;
+		}
+
+		public sealed override async Task RegisterCodeFixesAsync (CodeFixContext context)
+		{
+			var root = await context.Document.GetSyntaxRootAsync (context.CancellationToken).ConfigureAwait (false);
+
+			var diagnostic = context.Diagnostics.First ();
+			var diagnosticSpan = diagnostic.Location.SourceSpan;
+
+			SyntaxNode targetNode = root!.FindNode (diagnosticSpan);
+			CSharpSyntaxNode? declarationSyntax = CodeFixProviderOperations.FindAttributableParent (targetNode, CodeFixProviderOperations.AttributeableParentTargets.All);
+
+			if (declarationSyntax is not null) {
+				var semanticModel = await context.Document
+					.GetSemanticModelAsync (context.CancellationToken).ConfigureAwait (false);
+				var symbol = semanticModel!.Compilation.GetTypeByMetadataName (FullyQualifiedUnconditionalSuppressMessageAttribute);
+
+				// Register a code action that will invoke the fix.
+				context.RegisterCodeFix (
+					CodeAction.Create (
+						title: s_title,
+						createChangedDocument: c => AddUnconditionalSuppressMessage (
+							context.Document, root, declarationSyntax, diagnostic, symbol!, c),
+						equivalenceKey: s_title),
+					diagnostic);
+			}
+		}
+		private static async Task<Document> AddUnconditionalSuppressMessage (
+			Document document,
+			SyntaxNode root,
+			CSharpSyntaxNode containingDecl,
+			Diagnostic diagnostic,
+			ITypeSymbol UnconditionalSuppressMessageSymbol,
+			CancellationToken cancellationToken)
+		{
+			var editor = new SyntaxEditor (root, document.Project.Solution.Workspace);
+			var generator = editor.Generator;
+
+			var semanticModel = await document.GetSemanticModelAsync (cancellationToken).ConfigureAwait (false);
+			if (semanticModel is null) {
+				return document;
+			}
+
+			// UnconditionalSuppressMessage("Rule Category", "Rule Id", Justification = "<Pending>")
+			var category = generator.LiteralExpression (diagnostic.Descriptor.Category);
+			var categoryArgument = generator.AttributeArgument (category);
+
+			var title = diagnostic.Descriptor.Title.ToString (CultureInfo.CurrentUICulture);
+			var ruleIdText = string.IsNullOrWhiteSpace (title) ? diagnostic.Id : string.Format ("{0}:{1}", diagnostic.Id, title);
+			var ruleId = generator.LiteralExpression (ruleIdText);
+			var ruleIdArgument = generator.AttributeArgument (ruleId);
+
+			var justificationExpr = generator.LiteralExpression ("<Pending>");
+			var justificationArgument = generator.AttributeArgument ("Justification", justificationExpr);
+
+			SyntaxNode[] attrArgs = new[] { categoryArgument, ruleIdArgument, justificationArgument };
```

Comment:
This method also looks very similar, except for the syntax of the attribute arguments. You could add a new abstract base class `AddAttributeToParentCodeFixProvider`, and then have both of the code fixers derive from it. Then, have an abstract method that constructs the syntax for the arguments, and override it for each of the code fix providers.